pkg/backup/backup.go

```diff
@@ -420,3 +435,70 @@ func (*realItemBackupper) backupItem(ctx *backupContext, item map[string]interfa
 
 	return nil
 }
+
+func (ib *realItemBackupper) handlePodExecHooks(ctx *backupContext, namespace, name string, item, metadata map[string]interface{}) error {
+	for _, hookSpec := range ctx.backup.Spec.Hooks.Resources {
```

Comment:
I would be interested in having a conversation about the potential to add these as configuration options to the resources themselves and have that compliment the backup spec.  Instead of *strictly* using the `hooks` in the BackupSpec itself, I personally think a more user friendly apporach would be to have a set of annotations on the resources themselves.  

It gets rather ugly to encode things in the annotations via JSON ala. InitContainers when they were in alpha, so that's a drawback however it would be very handy as an administrator of an evolving K8 ecosystem to automatically allow new applications to be backed up "for free" in a proper way (i.e. I don't have to go in and change the backup spec just because a new service was deployed).

Reply:
@jrnt30 we can certainly discuss annotations. I don't want to encode json in them (and we're not doing that any more for alpha features in Kube), so I would probably want something like this (assuming we don't violate any validations):

```
hook.backup.ark.heptio.com/script = /usr/bin/myscript
hook.backup.ark.heptio.com/on-error = Fail
hook.backup.ark.heptio.com/timeout = 5m
```

Reply:
@jrnt30 what do you think should happen if there is a hook defined in both the backup spec and via an annotation?

Reply:
I'm tempted to say annotation wins

Reply:
I would say so as well.  That would allow a more general (making this up) *logging hook* for all pods and an explicit *redis hook* on a specific pod.

Reply:
I'm renaming `script` to `command` to be consistent with the terms in the backup spec. Required annotations will be container and command. on-error will default to fail. timeout will probably default to something small-ish like 30s.

Reply:
Sounds good.  This looks like a solid start that we could build off of for sure.